core/io/resource_uid.cpp

```diff
@@ -45,23 +45,44 @@ String ResourceUID::get_cache_file() {
 	return ProjectSettings::get_singleton()->get_project_data_path().path_join("uid_cache.bin");
 }
 
+static constexpr uint8_t uuid_characters[] = { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', '0', '1', '2', '3', '4', '5', '6', '7', '8' };
+static constexpr uint32_t uuid_characters_nb = (sizeof(uuid_characters) / sizeof(*(uuid_characters)));
+
 String ResourceUID::id_to_text(ID p_id) const {
 	if (p_id < 0) {
 		return "uid://<invalid>";
 	}
+	// Max 0x7FFFFFFFFFFFFFFF size is 19 digits and uid:// is 6 + 1 for null.
 	String txt;
-
+	txt.resize(26);
+
+	char32_t *p = txt.ptrw();
+	p[0] = 'u';
+	p[1] = 'i';
+	p[2] = 'd';
+	p[3] = ':';
+	p[4] = '/';
+	p[5] = '/';
```

Comment:
Hrm, would be nice if there was an easier way to do this... Maybe `std::copy_n("uid://", 6, p)`?

Reply:
Not really though, as one is a char (aka 1 byte) and the other is char32_t (aka 4 bytes). The copy routines want the same size on both sides. And anyway, this is so small that calling a function will be more trouble than 6 assembly **mov**

Reply:
It would _likely_ be inlined and then unrolled by the compiler, so my suggestion would simply be to make it more readable. But it's not really important :)

Reply:
If you know one function in Godot that can do this, I can replace this with it, no problem.